Sources/NIOPosix/SelectableEventLoop.swift

```diff
@@ -439,28 +601,44 @@ Further information:
                     while let sched = self._scheduledTasks.pop() {
                         scheduledTasksCopy.append(sched)
                     }
+                    immediateTasksCopy.reserveCapacity(self._immediateTasks.count)
+                    while let immediate = self._immediateTasks.popFirst() {
+                        immediateTasksCopy.append(immediate)
+                    }
```

Comment:
Instead of this `while let`, can we use `swap`?

Reply:
This is removing tasks from `self._immediateTasks` and putting them to a different array `immediateTasksCopy`. So `swap` isn't applicable, unless there is some other `swap` I'm not aware of.

Reply:
Ah, this [`swap`](https://developer.apple.com/documentation/swift/swap(_:_:)). Adopted here: https://github.com/apple/swift-nio/pull/2645/commits/8b818e6af83c1ba0808f4c4778eda6be09c3f98b#diff-565b354476ebe678da3be159a85e33e0b7630b1e45b62b20efe506210c1d62faR632